composition.js

```diff
@@ -0,0 +1,99 @@
+"use strict"
+const fs = require('fs')
+const options = JSON.parse(fs.readFileSync('./cookies.json','utf8'))
+const value = Object.values(options)
+const key = Object.keys(options)
+// console.log(value)
+class Cookie {
+    constructor(name,bahan) {
+
+        this.name = name
+        this.status = 'mentah'
+        this.ingredients = this.generateIngredients(bahan)
+        // console.log(this.ingredients)
+    }
+    bake() {
+        this.status = 'selesai dimasak'
+    }
+    generateIngredients(bahan) {
+        let temp = []
+        for (let key in bahan) {
+            temp.push(new Ingredients(key,bahan[key]))
+        }
+        // console.log(temp)
+        return temp
+
+        // this.ingredients.push(new Ingredients(namaBahan,jumlah))
+    }
+
+}
+
+class PeanutButtter extends Cookie {
+    constructor(name,bahan) {
```

Comment:
karena peanut butter namanya sudah pasti 'peanut butter' jadi tidak perlu terima parameter name lagi di constructor
```suggestion
    constructor(bahan) {
```